### ISSUE TITLE
After a Cluster Update there are always some keepalivedgroup Pods in CrashLoopBackOff

### ISSUE THREAD

**Issue by cobexer** [redhat-cop/keepalived-operator#24]
After a Cluster Update there are always some keepalivedgroup Pods in CrashLoopBackOff:
The log of the keepalived container shows:
```
Starting Keepalived v1.3.5 (03/19,2017), git commit v1.3.5-6-g6fa32f2
Opening file '/etc/keepalived.d/keepalived.conf'.
daemon is already running
```
The reason seems to be a stale PID file:
```
sh-4.4# ls -la /proc/*/exe
ls: cannot read symbolic link '/proc/35/exe': Permission denied
lrwxrwxrwx. 1 root root 0 Aug  1 07:40 /proc/1/exe -> /usr/bin/pod
lrwxrwxrwx. 1 root root 0 Aug  1 17:21 /proc/22/exe -> /usr/bin/bash
lrwxrwxrwx. 1 root root 0 Aug  1 17:18 /proc/28391/exe -> /usr/bin/bash
lrwxrwxrwx. 1 root root 0 Aug  1 17:27 /proc/28854/exe -> /usr/bin/coreutils
lrwxrwxrwx. 1 1001 root 0 Aug  1 07:40 /proc/35/exe
lrwxrwxrwx. 1 root root 0 Aug  1 17:27 /proc/self/exe -> /usr/bin/coreutils
lrwxrwxrwx. 1 root root 0 Aug  1 17:27 /proc/thread-self/exe -> /usr/bin/coreutils
sh-4.4# cat /etc/keepalived.pid/keepalived.pid 
12
```

I think keepalived needs to remove the pid file instead of using it to detect a running instance (which will never exist in this container scenario?!)

**Comment by David-Igou** [comment]
@cobexer Have you found any ways to recreate this other than a cluster update? I Was able to recreate this by updating an OCP cluster (z stream) the operator was running on. But that makes it difficult to debug..

**Comment by raffaelespazzoli** [comment]
@David-Igou deploying a machine config to the worker nodes machine config pool, should make recycle all of the nodes.

**Comment by raffaelespazzoli** [comment]
closed by #28
